tests/integration/targets/kea_command/tasks/main.yml

```
@@ -0,0 +1,138 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+---
+####################################################################
+# WARNING: These are designed specifically for Ansible tests       #
+# and should not be used as examples of how to write Ansible roles #
+####################################################################
+
+# Copyright © Thorsten Glaser <tglaser@b1-systems.de>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- name: Ensure this is run on the right distro and version the integration test was written for
+  ansible.builtin.assert:
+    that:
+      - ansible_distribution_major_version | int == 13
+      - ansible_distribution | lower == 'debian'
+    fail_msg: "This test was written for a Debian 13 (trixie) container due to uses of its default configuration"
+    quiet: true
```

Easy as that:
```suggestion
- name: Ensure this is run on the right distro and version the integration test was written for
  ansible.builtin.meta: end_play
  when: ansible_os_family != 'Debian' and ansible_distribution_major_version | int != 13
```

This way we can enable the test and, whenever Debian trixie is enabled in the CI, it will already work.

---

We won't have Debian Trixie as a VM, so it won't work :) But there are other OSes that we *do* have as a VM that should work already now (with potentially little changes, like Ubuntu 24.04).

---

Tje ansible-core testing infrastructure does not provide Debian **VMs**. We can run **containers** though (and we will run Debian Trixie containers).

The tests for this modlue do not run in containers, though.

---

Ow. If they don’t run in containers, there’s little chance I can do the things needed to start the KEA services in there.

---

@felixfontein I gave the thumbs up to your last comment, but then later it hit me: why? why won't the tests run in containers? I am not familiar with KEA, so I am assuming something in this tool that prevents that from happening? Glancing over the test code I could not see anything else that would justify that - but I really just glanced quickly, I may have missed something.

---

@russoz the tests set up a new networking device, that isn't supported in "normal" containers (without additional capabilities, or privileged mode): https://github.com/ansible-collections/community.general/pull/10709#issuecomment-3218206056

Since we want that containerized tests "just work" if you run `ansible-test integration --docker <right image> <test name>`, without having to supply specific additional options to give more capabilities to the container (and especially without making it a privileged container), it won't run in CI this way.